tests/test_atomic_metadata_extractor.py

```diff
@@ -0,0 +1,45 @@
+import pytest  # noqa
+from colonel.utils.discovery import AtomicMetadataExtractor, MetadataParsingException, AtomicMetadata
+from colonel.models import InPort, OutPort
+
+
+def test_no_metadata_fails():
+    with pytest.raises(MetadataParsingException):
+        source = "void main(char** argv) {\n\tprintf(\"\");\n}"
+        AtomicMetadataExtractor(source).extract()
+
+
+def test_just_model_name_in_metadata():
+    source = """
+    @ModelMetadata
+    name:perro
+    """
+    assert AtomicMetadataExtractor(source).extract() == AtomicMetadata("perro", [], [])
```

Comment:
Related to the comment before, you're assuming you're inside a comment only because the input is a string. Same for the other tests.